tests/modules/nf-core/quilt/quilt/main.nf

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { QUILT_QUILT                   } from '../../../../../modules/nf-core/quilt/quilt/main.nf'
+include { QUILT_QUILT as QUILT_OPTIONAL } from '../../../../../modules/nf-core/quilt/quilt/main.nf'
+
+
+    // input sequencing data (bam)
+    bam = [
+        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/quilt/NA12878.haplotagged.1.0.bam', checkIfExists: true),
+        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/quilt/NA12878.ont.1.0.bam',         checkIfExists: true),
+        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/quilt/NA12878.illumina.1.0.bam',    checkIfExists: true)
+    ]
+
+    bai = [
+        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/quilt/NA12878.haplotagged.1.0.bam.bai', checkIfExists: true),
+        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/quilt/NA12878.ont.1.0.bam.bai',         checkIfExists: true),
+        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/quilt/NA12878.illumina.1.0.bam.bai',    checkIfExists: true)
+    ]
+
+    bamlist = [
+        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/quilt/bamlist.1.0.txt',                checkIfExists: true)
+    ]
+
+    bam_bai_bamlist = [ [ id:"test", chr:"chr20" ], bam, bai, bamlist ]
+
+    // input reference data
+
+    reference_haplotype_file = [
+        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/quilt/ALL.chr20_GRCh38.genotypes.20170504.chr20.2000001.2100000.noNA12878.hap.gz', checkIfExists: true)
+    ]
+
+    reference_legend_file = [
+        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/quilt/ALL.chr20_GRCh38.genotypes.20170504.chr20.2000001.2100000.noNA12878.legend.gz', checkIfExists: true)
+    ]
+
+    genetic_map_file = [
+        file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/quilt/CEU-chr20-final.b38.txt.gz', checkIfExists: true)
+    ]
+
+    // parameters
+
+    def chr = "chr20"
```

Comment:
Why are these parameters all being set in value channels? They feel like they should just be being supplied in ext.args in general, or could be given as part of the meta.map.

Reply:
The decision to set these parameters in value channels serves two key reasons. Firstly, it aligns with the design and functionality of other nf-core modules, such as [STITCH](https://github.com/nf-core/modules/blob/master/modules/nf-core/stitch/main.nf), which is a tool developed by the same author. We wanted to have consistency in design that would facilitate ease of use and familiarity for users. 

Secondly, these parameters cannot be supplied via ext.args because the chromosome and chunks are dynamically read from the channel output of another tool, specifically, the "glimpse_chunk" tool. This approach allows for the simultaneous launch of several parallel processes, each handling a distinct section of the genome. However, I agree with you that some parameters (buffer, nGen, seed) could be supplied via ext.args and I am working on this implementation.